lib/edda-server/src/change_set_processor_task/materialized_view.rs

```diff
@@ -239,74 +250,141 @@ async fn build_mv_inner(
         for change in changes {
             for &mv_kind in &independent_mvs {
                 match mv_kind {
-                    ReferenceKind::View => {
-                        let mv_id = change.entity_id.to_string();
-
-                        let (maybe_patch, maybe_frontend_object) =
-                            si_frontend_types_macros::build_mv!(
-                                ctx,
-                                frigg,
-                                change,
-                                mv_id,
-                                si_frontend_types::view::View,
-                                dal::diagram::view::View::as_frontend_type(
+                    ReferenceKind::ActionPrototypeViewsByComponentId => {
+                        if si_frontend_types::action::ActionPrototypeViewsByComponentId::trigger_entity() == change.entity_kind {
+                            let mut component_ids = HashSet::new();
+                            for maybe_action_change in changes {
+                                match maybe_action_change.entity_kind {
+                                    EntityKind::Action => {
+                                        match Action::component_id(ctx, maybe_action_change.entity_id.into_inner().into()).await {
+                                            Ok(Some(component_id)) => {
+                                                component_ids.insert(component_id);
+                                            }
+                                            Ok(None) => {}
+                                            Err(err) => error!(si.error.message = ?err, "error getting component for action")
+                                        }
+                                    },
+                                    EntityKind::ActionPrototype =>  match ActionPrototype::schema_variant_id(ctx, maybe_action_change.entity_id.into_inner().into()).await {
+                                        Ok(schema_variant_id) => match SchemaVariant::list_component_ids(ctx, schema_variant_id).await {
+                                            Ok(component_ids_for_prototype) => component_ids.extend(component_ids_for_prototype),
+                                            Err(err) => error!(si.error.message = ?err, "error getting components for schema variant")
+                                        }
+                                        Err(err) => error!(si.error.message = ?err, "error getting schema variant for action prototype")
+                                    },
+                                    _ => {}
+                                }
+                            }
```

Comment:
@jhelwig @fnichol @britmyerss @jobelenus this is what we discussed about partial completion vs. error and eject. I don't think our work is done here, but is probably done for the purposes of this PR.